ddd-koans/Value Object/Tiny Types/src/Main.kt

```diff
@@ -0,0 +1,37 @@
+sealed class TimeSignature {
+    companion object {
+        private val timeSignatureRegex = "^(\\d{1,2})/(\\d{1,2})$".toRegex()
+
+        fun create(numerator: Int, denominator: Int): TimeSignature {
+            return runCatching {
+                ValidTimeSignature(Numerator(numerator), Denominator(denominator))
+            }.getOrDefault(InvalidTimeSignature)
+        }
+
+        fun of(timeSignature: String): TimeSignature {
+            val groupValues = timeSignatureRegex.matchEntire(timeSignature)?.groupValues
+            return if (groupValues == null) {
```

Comment:
it could be replaced with `groupValues?.let{...}` the question is more about taste and readability

Reply:
I considered it but left it as it is. But I'll change it to `?.let` version with pleasure.